src/components/ViewList.js

```diff
@@ -1,68 +1,56 @@
 import { Link } from 'react-router-dom';
+import calculateEstimate from '../lib/estimates';
 
 const ViewList = () => {
-
   // If the list is empty, add a prompt and link to Add Items
   let currentList = useContext(ListContext);
-  let token = currentList.token;
-
-  const [itemsPurchased, setItemsPurchased] = useState({});
 
   const handleCheck = async (e) => {
-    e.persist();
-    const item = e.target.name;
-    const isChecked = e.target.checked;
-    await setItemsPurchased((prevItemsPurchased) => ({
-      ...prevItemsPurchased,
-      [item]: isChecked,
-    }));
-    updateDatabase(e);
-  };
+    const timeNow = new Date().getTime() / 1000;
+    // retrieve event informationt
+    let docId = e.target.id;
+    let currentRef = currentList.itemsRef.doc(docId);
 
-  useEffect(() => {
-    const handleTiming = () => {
-      let currentTime = Date.now();
-      console.log(currentTime);
+    //find context for current item
+    const currentItem = currentList.userList.find((item) => {
+      return item.id == docId;
+    });
 
-      const lastPurchasedTimeArray =
-        currentList.userList &&
-        currentList.userList.map((element) => {
-          const container = {};
-          container.itemName = element.itemName;
-          container.t = element.lastPurchased;
-          return container;
-        });
+    // store current item data
+    let numberOfPurchases = 1;
+    currentItem.numberOfPurchases
+      ? (numberOfPurchases = currentItem.numberOfPurchases + 1)
+      : (numberOfPurchases = 1);
+    let lastEstimate =
+      parseInt(currentItem.lastEstimate) || parseInt(currentItem.timeFrame);
+    let latestInterval;
+    let lastPurchased;
 
-      lastPurchasedTimeArray &&
-        lastPurchasedTimeArray.forEach((element) => {
-          if (element.t + 86400000 < currentTime) {
-            setItemsPurchased((prevItemsPurchased) => ({
-              ...prevItemsPurchased,
-              [element.itemName]: false,
-            }));
-          } else
-            setItemsPurchased((prevItemsPurchased) => ({
-              ...prevItemsPurchased,
-              [element.itemName]: true,
-            }));
-        });
-    };
-    handleTiming();
-  }, []);
+    // if item has been purchased more than two times, calculate the last interval, otherwise set to original timeframe
+    if (numberOfPurchases > 2) {
+      lastPurchased = currentItem.lastPurchased.seconds;
+      latestInterval = Math.ceil((timeNow - lastPurchased) / (24 * 60 * 60));
+    } else {
+      latestInterval =
+        parseInt(currentItem.timeFrame) || parseInt(currentItem.lastEstimate);
+    }
 
-  const updateDatabase = (e) => {
-    let docId = e.target.id;
-    console.log(docId);
+    lastEstimate = calculateEstimate(
+      lastEstimate,
+      latestInterval,
+      numberOfPurchases,
+    );
 
-    let currentRef = currentList.itemsRef.doc(docId);
-    console.log(currentRef);
+    console.log('the calculated interval is ', lastEstimate);
```

Comment:
Love that you made it easy for us to see what was happening in the background. Best practice though is to remove console.log statements so we don't clutter up our console. Perhaps instead of the console.log's we could get instructions on what database changes we should look for.